puts/cmake/CheckCCompilerCanPrintDeps.cmake

```diff
@@ -0,0 +1,21 @@
+include_guard(GLOBAL)
```

Comment:
This file needs some comment what it is and how to use it. Not a CMake pro myself.

Reply:
It is a helper cmake script to check that the compiler supports the `-M` option.

I have taken note to add some comments to the cmake files in a future PR. I'd rather wait for the entire build chain to be in place, because things might move around a bit. Tell me if you absolutely want it in this PR.